src/rvsdg/egglog_optimizer/interval-analysis.egg

```diff
@@ -65,3 +76,74 @@
         (union gamma (OperandGroup (SubstVecOperandAll (vec-get outputs 0) inputs)))
       )
 )
+
+; HACK: Table of fucking integers
+(datatype FNum
+  (NumN i64))
+(NumN 0)
+(NumN 1)
+(NumN 2)
+(NumN 3)
+
+; push interval into Gamma
+(rule (
+      (= gamma (Gamma pred (VO ins) (VVO outs)))
+      (NumN n)
+      (= arg (vec-get ins n))
+      (= (IntI l h) (ival arg))
+      )
+      ((set (context-ival (Arg n) gamma) (IntI l h)))
+)
+
+; interval of a + b = (la + lb, ha + hb)
```

Comment:
@oflatt it's pretty painful to have 4 versions of this rule- is there a way to match on "Either an ival or a context-ival"?

Reply:
One way to do it would be to have another table called `either-ival-or-context-ival` and a rule that adds to that table. I think that would be reasonable 
Another strategy is to generate these rules from rust.

Reply:
What would the type of this table be? Are there `Option`s?